src/tiled/tileseteditor.cpp

```diff
@@ -826,6 +861,40 @@ void TilesetEditor::removeTerrainType()
         undoStack->endMacro();
 }
 
+void TilesetEditor::currentWangSetChanged(const WangSet *wangSet)
+{
+    TilesetView *view = currentTilesetView();
+    if (!view)
+        return;
+
+    if (wangSet)
+        view->setWangSet(wangSet);
+}
+
+void TilesetEditor::addWangSet()
+{
+    Tileset *tileset = currentTileset();
+    if(!tileset)
+        return;
+
+    //2 and 0 are default values for number of edges and corners TODO define this some where better?
+    WangSet *wangSet = new WangSet(tileset, 2, 0, QString(), -1);
+    wangSet->setName(tr("New Wang Set"));
+
+    mCurrentTilesetDocument->undoStack()->push(new AddWangSet(mCurrentTilesetDocument,
+                                                              wangSet));
+
+    mWangDock->editWangSetName(wangSet);
+}
+
+void TilesetEditor::removeWangSet()
+{
+    WangSet *wangSet = mWangDock->currentWangSet();
```

Comment:
Please do check that there is a current wang set, even if the button to trigger this action should be disabled at this point.